fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -888,20 +888,23 @@ public CommandProcessingResult adjustLoanTransaction(final Long loanId, final Lo
         }
 
         final String noteText = command.stringValueOfParameterNamed("note");
-        if (StringUtils.isNotBlank(noteText)) {
-            changes.put("note", noteText);
-            Note note = null;
-            /**
-             * If a new transaction is not created, associate note with the
-             * transaction to be adjusted
-             **/
-            if (newTransactionDetail.isGreaterThanZero(loan.getPrincpal().getCurrency())) {
-                note = Note.loanTransactionNote(loan, newTransactionDetail, noteText);
-            } else {
-                note = Note.loanTransactionNote(loan, transactionToAdjust, noteText);
-            }
-            this.noteRepository.save(note);
-        }
+        changes.put("note", noteText);  
+         Note note = null;  
+         /**  
+          * If a new transaction is not created, associate note with the  
+          * transaction to be adjusted  
+          **/  
+         if (newTransactionDetail.isGreaterThanZero(loan.getPrincpal().getCurrency())) {  
+             note = Note.loanTransactionNote(loan, newTransactionDetail, noteText);  
+         } else {  
+            note = this.noteRepository.findByLoanTransactionId(transactionId);  
```

Comment:
This API cannot return a single note object... so premise of this solution becomes invalid.... What is the expectation of this PR? can you please explain or point us to relevant JIRA issue?